psi4/driver/p4util/text.py

```diff
@@ -244,22 +244,26 @@ def message_box(message: str = None, max_width: int = 80):
        box containing the message as a multiline string
     """
     from textwrap import wrap
-    
-    is_long = (len(message) > max_width)
-    box_width = max_width if is_long else len(message)
 
-    message = " ".join(message.split("\n"))
-    message = wrap(message, box_width, break_long_words=False)
+    # ensure box is within min/max boundaries
+    msg = message.splitlines()
+    max_line = len(max(msg, key=len))
+    is_long = (max_line > max_width)
+    is_short = (max_line < min_width)
+    box_min = min_width if is_short else max_line
+    box_width = max_width if is_long else box_min
```

Comment:
I don't swear I've got the logic all right, but could be compacted with min/max perhaps: `box_width = max(min(max_width, max_line), min_width)`.

Reply:
oh wow, hmm, yes that should work.